src/darwin/Framework/CHIP/MTRDevice_Concrete.mm

```diff
@@ -350,6 +350,14 @@ @implementation MTRDevice_Concrete {
     NSDate * _Nullable _mostRecentReportTimeForDescription;
     // Copy of _lastSubscriptionFailureTime that is safe to use in description.
     NSDate * _Nullable _lastSubscriptionFailureTimeForDescription;
+    // Copy of _state that is safe to use in description.
+    MTRDeviceState _deviceStateForDescription;
```

Comment:
In what sense is this ForDescription?  We are not using it in description....

We take the description lock in `_internalProperties` when we are working with the things that are in fact protected by that lock.  It looks to me like we should just take our normal lock there when we need things like state, estimated start time, etc....

Description is special because we might need to get it while holding our normal lock, hence its separate lock.  But would we be calling `_internalProperties` while holding `_lock`?  Looks to me like we are always calling it from an async call on some delegate queue, right?  So just taking `_lock_ and getting the data we want should be fine.

I don't think most of the changes to this file are needed.

Reply:
@bzbarsky-apple Let's follow up here in an issue.

Reply:
No, my point is we should not be making these changes.  I'll go ahead and just fix this part up....